sdk/identity/azure-identity-extensions/src/test/java/com/azure/identity/extensions/implementation/credential/provider/CachingTokenCredentialProviderTest.java

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.extensions.implementation.credential.provider;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.identity.DefaultAzureCredential;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class CachingTokenCredentialProviderTest {
+
+    @Test
+    void cacheTokenCredential() {
```

Comment:
Let's add three more cases:
- One for using TokenCredentialProvider.get(Options) to get token credentials, but the two options can be resolved to one key
- One for using both `TokenCredentialProvider.get()` and `TokenCredentialProvider.get(Options)` to get token credentials, and the default options and the second options can be resolved to one key
- One for different options cache key, will return two token credentials